### ISSUE TITLE
Error when using Mermaid compact display mode

### ISSUE THREAD

**Issue by kpspu** [laurent22/joplin#8701]
Experiencing with all Joplin versions. I think the problem in old Mermaid version.

Got an Mermaid error when type in note:

```replaced-mermaid-header-to-prevent-drawing

--- 
displayMode: compact 
---

gantt
    title A Gantt Diagram
    dateFormat  YYYY-MM-DD

    section Section
    A task           :a1, 2014-01-01, 30d
    Another task     :a2, 2014-01-20, 25d
    Another one      :a3, 2014-02-10, 20d

```

Should work as here: https://mermaid.js.org/syntax/gantt.html#output-in-compact-mode

**Comment by StanislawHorna** [comment]
+1

**Comment by tessus** [comment]
Fixed by #8890
